### PR TITLE
Support load-balanced CP chunk masks

### PR DESCRIPTION
**Pull request by huytransformer**
# Description

Enables `attention_type=chunk` with load-balanced CP by making the chunk mask use the same query ordering as the load-balanced causal mask.

This PR:
- adds `LoadBalancedChunkedCausalMask` for Splash attention
- uses `segment_positions` for dot-product chunk and local-sliding masks under load-balanced CP

FIXES: b/435512699

# Tests

Ran:

```bash
pyink --check --pyink-indentation=2 --line-length=122 \
  src/maxtext/layers/attention_op.py \
  tests/unit/attention_test.py \
  tests/unit/configs_value_test.py \
  src/maxtext/configs/pyconfig_deprecated.py

python3 -m pytest tests/unit/attention_test.py -k 'load_balanced_chunk_window or dot_product_chunk_mask_uses_segment_positions or dot_product_local_mask_uses_segment_positions or context_parallel_chunk_attention_rejected'

python3 -m pytest tests/unit/configs_value_test.py::ConfigTest::test_load_balanced_chunk_context_parallel_config

python3 -m maxtext.trainers.pre_train.train src/maxtext/configs/base.yml \
  run_name=lb_cp_chunk_e2e \
  model_name=llama3.1-8b \
  dataset_type=synthetic \
  steps=1 \
  enable_checkpointing=false \
  packing=false \
  attention=flash \
  attention_type=chunk \
  chunk_attn_window_size=256 \
  context_parallel_load_balance=True \
  ici_context_parallelism=4 \
  ici_fsdp_parallelism=16 \
  per_device_batch_size=0.25 \
  max_target_length=2048

on v5p-128 with EXIT_CODE=0.
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by gobbleturk — APPROVED** [review]

File: src/maxtext/layers/attention_op.py [hunk 1]
```diff
@@ -710,20 +714,33 @@ def generate_attention_mask(
     elif model_mode == MODEL_MODE_AUTOREGRESSIVE and q_seq_len == 1:
       # In autoregression, the query position is the last position in the KV sequence.
       next_pos = kv_seq_len - 1
+    use_segment_positions = (
+        segment_positions is not None
+        and self.config.context_parallel_load_balance
+        and self.mesh.shape[self.config.context_sharding] > 1
```
Comment (NuojCheng) [inline src/maxtext/layers/attention_op.py]: nit: in future when we enable both CP and SP(Ulysses CP with diff physical axis), we might need to change the logic of using `self.mesh.shape[self.config.context_sharding]` for fetching seq parallelism size.

Comment (NuojCheng) [reply src/maxtext/layers/attention_op.py]: Let's update 
`self.mesh.shape[self.config.context_sharding]`
to
`self.mesh.shape.get(self.config.context_sharding, 1)`
for robustness.

File: src/maxtext/configs/pyconfig_deprecated.py [hunk 2]
```diff
@@ -250,9 +250,6 @@ def validate_keys(keys):
     )
 
   context_parallel_size = get_context_parallel_size(keys)
-  # TODO remove after b/435512699 resolved
-  if context_parallel_size > 1 and keys["context_parallel_load_balance"] and keys["attention_type"] == "chunk":
-    raise ValueError("Currently load-balanced context parallelism is not supported for chunk attention.")
```
Comment (NuojCheng) [inline src/maxtext/configs/pyconfig_deprecated.py:255]: no need to update. This file is deprecated

**Review by NuojCheng — APPROVED** [review]
